### ASSIGNMENT- Basic Stat-2
##### Estimation And Confidence Intervals

In [185]:
import numpy as np
import scipy.stats as stats

In [67]:
sample=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
n=15

a. Build 99% Confidence Interval Using Sample Standard Deviation.
Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.


In [118]:
x_bar=np.mean(sample)

In [120]:
x_bar

1.2386666666666666

In [122]:
std=np.std(sample,ddof=1)
std

0.19316412956959936

In [59]:
## Sample mean(x_bar)=1.23
## Sample standard deviation(std)=.1931
## sample size(n)=15
## degree of freedom=15-1
## confidence interval=99%=.99
## alpha=1-.99=.01

In [124]:
## t_critical=stats.t.ppf(1-alpha/2,degree_of_freedom)
t_critical=stats.t.ppf(1-(.01/2),14)

In [126]:
t_critical

2.976842734370834

In [154]:
## pop_lower_mean=sample_mean-t_critical*(sample_standard_deviation/sqrt_of_sample_size)
pop_lower_mean=x_bar-t_critical*(std/np.sqrt(n))
pop_lower_mean

1.0901973384384906

In [156]:
## pop_lower_mean=sample_mean+t_critical*(sample_standard_deviation/sqrt_of_sample_size)
pop_upper_mean=x_bar+t_critical*(std/np.sqrt(n))
pop_upper_mean

1.3871359948948425

###### Rationale for Using the t-Distribution
The t-distribution is applied because:

Small Sample Size (n = 15): With a sample size smaller than 30, the Central Limit Theorem doesn't assure that the sample mean's distribution will be normal. Therefore, the t-distribution is a more suitable choice for small sample sizes.

Unknown Population Standard Deviation: Since the population standard deviation is unknown and estimated from the sample data, the t-distribution is needed to account for the additional uncertainty in this estimation.

Hence, the t-distribution is the appropriate choice for constructing the confidence interval in this case.

##### So, the 99% confidence interval for the mean number of characters printed before the print-head fails is (1.0901, 1.3871).
##### population_lower_mean=1.0901
##### population_upper_mean=1.3871

##### b. Build 99% Confidence Interval Using Known Population Standard Deviation.If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.


In [162]:
STD=.2

In [168]:
## Sample mean(x_bar)=1.23
## Population standard deviation(STD)=.2
## sample size(n)=15
## degree of freedom=15-1
## confidence interval=99%=.99
## alpha=1-.99=.01


In [140]:
z_critical=stats.norm.ppf(1-(.01/2))

In [142]:
z_critical

2.5758293035489004

In [175]:
## pop_lower_mean=sample_mean-z_critical*(population_standard_deviation/sqrt_of_sample_size)
pop_lower_mean=x_bar-z_critical*(STD/np.sqrt(n))
pop_lower_mean

1.1056514133957607

In [177]:
## pop_lower_mean=sample_mean+z_critical*(population_standard_deviation/sqrt_of_sample_size)
pop_upper_mean=x_bar+z_critical*(STD/np.sqrt(n))
pop_upper_mean

1.3716819199375725

#### Conclusion
population_lower_mean=1.1056

population_upper_mean=1.3716